File: src/main/java/example/inflearn_starter/chap2/lect8/FunctionTest.java
package example.inflearn_starter.chap2.lect8;

public class FunctionTest {
    public static void main(String[] args) {
        printAll("Hello", "World", "!");
    }

    public int max(int a, int b) {
        if(a > b) return a;
        return b;
    }

    public static void repeat(String str, int num, boolean useNewLine) {
        for (int i = 1; i <= num; i++) {
            if (useNewLine) {
                System.out.println(str);
            } else {
                System.out.print(str);
            }
        }
    }

    public void repeat(String str, int num) {
        repeat(str, num, true);
    }

    public void repeat(String str) {
        repeat(str, 3, true);
    }

    public static void printAll(String... strings) {
        for (String string : strings) {
            System.out.println(string);
        }
    }
}

File: src/main/kotlin/example/inflearn_starter/chap2/lect8/FunctionTest.kt
package example.inflearn_starter.chap2.lect8

fun main() {
    repaet("Hello")
    repaet("Hello", 3)
    repaet("Hello", 3, false)

    repaet("Hello", useNewLine = false)

    printNameAndGender("Female", "최홍만")
    printNameAndGender(gender = "Female", name = "최홍만")
//    FunctionTest.repeat(str = "Hello", 1, false)

    printAll("Hello", "World", "!")

    val arr = arrayOf("Hello", "World", "!")
    printAll(*arr)
}

fun max(a: Int, b: Int): Int = if (a > b) a else b

fun repaet(
    str: String,
    num: Int = 3,
    useNewLine: Boolean = true
) {
    for (i in 1..num) {
        if (useNewLine) {
            println(str)
        } else {
            print(str)
        }
    }
}

fun printNameAndGender(name: String, gender: String) {
    println(name)
    println(gender)
}

fun printAll(vararg strings: String) {
    for (str in strings) {
        println(str)
    }
}